File: php/Creational_Patterns/simplefactory.php
<?php
class Factory{

  public static function NewApi($name=""){
      switch ( $name ) {
          case 'oms':
            return (new Oms);
            break;
          case 'wms':
            return (new Wms);
            break;
          default:
            return (new Oms)
            break;
      }
  }
}


interface IApi{
  function Say($name="");
}

class OmsApi implements IApi{
    function Say($name=""){
        echo "oms,".$name;
    }
}

class WmsApi implements IApi{
    function Say($name=""){
        echo "wms,".$name;
    }
}

//测试
$oms = Factory::NewApi('oms');
$oms->Say('felix');

$oms = Factory::NewApi('wms');
$oms->Say('felix');

File: golang/Creational_Patterns/simplefactory.go
/**
 * 简单工厂
 */
package simplefactory

import "fmt"


//API is an interface
type API interface{
  Say(name string) string
}

//omsApi implements the Api interface
type  omsApi struct{}

func(*omsApi) Say(name string) string{
  return fmt.Sprintf("oms,%s",name)
}

//wmsApi  implemnets the Api interface
type wmsApi struct{}

func (*wmsApi)Say(name string) string{
  return fmt.Sprintf("wms,%s",name)
}

func NewApi(t string) API{
  switch t{
    case "oms":
      return &omsApi{}
    case "wms":
      return &wmsApi{}
    default:
      return nil
  }
}
